### PR TITLE
add basic projectile

### PR DESCRIPTION
**Pull request by przemo1232**

### CODE REVIEW

**Review by AWarmCupOfTea — CHANGES_REQUESTED** [review]
Please format the code according to https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/gdscript_styleguide.html

**Review by AWarmCupOfTea — CHANGES_REQUESTED** [review]

File: scenes/player/player.gd [hunk 1]
```diff
@@ -45,6 +52,17 @@ func _physics_process(delta: float):
 	]
 
 
+func fire_projecile():
+	if PROJECTILE:
```
[the remark this answers is not in the document]
Comment (przemo1232) [reply scenes/player/player.gd:56]: i don't think so, but it was written like that in a tutorial so i added it

**Review by AWarmCupOfTea — CHANGES_REQUESTED** [review]

File: scenes/player/player.gd [hunk 2]
```diff
@@ -45,6 +52,17 @@ func _physics_process(delta: float):
...
+		var projectile = PROJECTILE.instance()
+		projectile.player_velocity = _velocity
+		get_tree().current_scene.add_child(projectile)
+		projectile.global_position = self.global_position
+		var rotation = self.global_position.direction_to(get_global_mouse_position()).angle()
+		projectile.rotation = rotation
+		last_attack_time = 0.0
```
[the remark this answers is not in the document]
Comment (przemo1232) [reply scenes/player/player.gd:63]: from what i found you can't pass parameters to a scene and this was the other option

[the remark this answers is not in the document]
Comment (AWarmCupOfTea) [reply scenes/player/player.gd:63]: https://godotengine.org/qa/4786/how-to-pass-parameters-when-instantiating-a-node

[the remark this answers is not in the document]
Comment (przemo1232) [reply scenes/player/player.gd]: moved the logic to the projectile scene and reformatted the code, hope it's good now

**Review by AWarmCupOfTea — APPROVED** [review]

**Pull request merged**